error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-with-flag.rs:30:11
   |
30 | #[ensures(!(x.model().a == x.model().b))]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-with-flag.rs:31:1
   |
31 | fn test1(x: X) {}
   | ^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: 5,
            b: 5,
        }
  --> $DIR/model-with-flag.rs:31:10
   |
31 | fn test1(x: X) {}
   |          ^
note: counterexample for "x"
       value:   X {
            a: A {
                a: 5,
            },
            b: B {
                b: 5,
            },
        }
  --> $DIR/model-with-flag.rs:31:10
   |
31 | fn test1(x: X) {}
   |          ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-with-flag.rs:31:16
   |
31 | fn test1(x: X) {}
   |                ^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/model-with-flag.rs:38:11
   |
38 | #[ensures(x.model().b == y.model().a)]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/model-with-flag.rs:39:1
   |
39 | fn test2(x: X, y: X) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: counterexample for "x"
       value:   X_model {
            a: 2,
            b: 5,
        }
  --> $DIR/model-with-flag.rs:39:10
   |
39 | fn test2(x: X, y: X) {}
   |          ^
note: counterexample for "x"
       value:   X {
            a: A {
                a: 2,
            },
            b: B {
                b: 5,
            },
        }
  --> $DIR/model-with-flag.rs:39:10
   |
39 | fn test2(x: X, y: X) {}
   |          ^
note: counterexample for "y"
       value:   X_model {
            a: 6,
            b: 2,
        }
  --> $DIR/model-with-flag.rs:39:16
   |
39 | fn test2(x: X, y: X) {}
   |                ^
note: counterexample for "y"
       value:   X {
            a: A {
                a: 6,
            },
            b: B {
                b: 2,
            },
        }
  --> $DIR/model-with-flag.rs:39:16
   |
39 | fn test2(x: X, y: X) {}
   |                ^
note: counterexample for "result"
       value:   ()
  --> $DIR/model-with-flag.rs:39:22
   |
39 | fn test2(x: X, y: X) {}
   |                      ^^

error: aborting due to 2 previous errors

